### PR TITLE
[Merged by Bors] - Add a flag to always use payloads from builders

### PR DESCRIPTION
**Pull request by danielrachi1**
## Issue Addressed

#4040 

## Proposed Changes

- Add the `always_prefer_builder_payload`  field to `Config` in `beacon_node/client/src/config.rs`.
- Add that same field to `Inner` in `beacon_node/execution_layer/src/lib.rs`
- Modify the logic for picking the payload in `beacon_node/execution_layer/src/lib.rs`
- Add the `always-prefer-builder-payload` flag to the beacon node CLI
- Test the new flags in `lighthouse/tests/beacon_node.rs`

### CODE REVIEW

**Comment by michaelsproul** [comment]
Ah that overflow error isn't the fault of your code. Are you running an older compiler version? If you update it should go away

File: lighthouse/src/main.rs [hunk 1]
```diff
@@ -312,6 +312,24 @@ fn main() {
...
+            builder's payload.")
+            .takes_value(true)
+            .global(true)
+        )
+        .arg(
+            Arg::with_name("always-prefer-builder-payload")
+            .long("always-prefer-builder-payload")
+            .help("If set, the beacon node always uses the payload from the builder instead of the local payload.")
+            .value_name("BOOL")
+            .takes_value(true)
```
Comment (paulhauner) [inline lighthouse/src/main.rs]:
```suggestion
```

The flag doesn't take a value (i.e. it's just `--flag` rather than `--flag value`) so we don't need these.

File: lighthouse/tests/beacon_node.rs [hunk 2]
```diff
@@ -340,6 +340,14 @@ fn trusted_peers_flag() {
         });
 }
 
+#[test]
```
Comment (paulhauner) [inline lighthouse/tests/beacon_node.rs:343]: Could we please add another test that doesn't have any flag provided and asserts `!config.always_prefer_builder_payload` 🙏

**Comment by divagant-martian** [comment]
@paulhauner has some thoughts on how we should do this changes, but to tackle the compiler error please apply this patch:
```diff
diff --git a/beacon_node/tests/test.rs b/beacon_node/tests/test.rs
index 1c11a8349..8ccb260d2 100644
--- a/beacon_node/tests/test.rs
+++ b/beacon_node/tests/test.rs
@@ -1,5 +1,5 @@
 #![cfg(test)]
-#![recursion_limit = "256"]
+#![recursion_limit = "512"]
 
 use beacon_chain::StateSkipConfig;
 use node_test_rig::{

```

**Review by paulhauner — CHANGES_REQUESTED** [review]
Thanks for the contribution @danielrachi 🙏

I appreciate the ambition with the `buildier-profit-margin` flag, however I'd like to request that we just do the `always_prefer_builder_payload` flag in this PR. We've already started testing this release so anything we add now should be as minimal as possible, both from a complexity/risk perspective and a time-to-implement-and-review perspective. We haven't had any user requests for the profit margin, so I think it's reasonable for us to omit it from this release.

As pointed out by @michaelsproul, it may be more user friendly to express the profit margins as a percentage rather than a wei value (e.g., "use the builder payload only if it's 50% more profitable). We can think about this after we get the release out, if you're keen to implement it then we'd love to have you involved. No pressure, of course.

So, in summary, it would be awesome if you could please adjust the `recursion_limit` (as per @divagant-martian's [diff/comment](https://github.com/sigp/lighthouse/pull/4052#issuecomment-1455932522)) and reduce the scope of this PR so that we can land this release nice and easily ☺️

**Review by paulhauner — APPROVED** [review]
Great, thanks @danielrachi!

In the interests of a quick resolution, I pushed a [commit](https://github.com/sigp/lighthouse/pull/4052/commits/ef5565b1c1cae63008c36ba7770f4f705892e06d) to:

- Move the CLI args to be underneath the `lighthouse bn` command, rather than at the root `lighthouse` level.
- Make the flag conflict with `--builder-profit-threshold`, since that flag could cause us to actually prefer a local payload.

**Review by michaelsproul — APPROVED** [review]
Nice! Let's merge, thanks @danielrachi
